framework/src/modules/dynamic_rewards/module.ts

```diff
@@ -94,6 +95,11 @@ export class DynamicRewardModule extends BaseModule {
 					request: getDefaultRewardAtHeightRequestSchema,
 					response: getDefaultRewardAtHeightResponseSchema,
 				},
+				{
+					name: this.endpoint.getInflationRate.name,
+					request: getDefaultRewardAtHeightRequestSchema,
```

Comment:
This endpoint does not take any input arguments, so metadata should not contain a request property... especially not from a different endpoint 🙂 

Probably a copy/paste mistake.

Reply:
this endpoint takes the `height` as input, so the schema-wise it's correct. Maybe, schema could be aliased to reflect the name properly

Although issue doesn't say about the input, i thought having option to specify height would be nice to get the future inflation rate as well. (i thought that was the proposal)

Reply:
The input height is needed to determine the current reward bracket. The name `getDefaultRewardAtHeightRequestSchema` is unchanged as no new schema is needed however I am open to suggestions for a neutral name that fits both endpoints.

Reply:
While reviewing this endpoint I was looking for `context.params` but I wasn't careful enough to notice that this endpoint simply passes on the whole context to `getDefaultRewardAtHeight`. I think this is the only endpoint I've seen that calls another endpoint?

But in that case, the requirements of quest #7799 are not correct, as they specify `{}` as endpoint request 🤔

Passing the whole context to another endpoint means that both endpoints use the same request schema. I find the aliasing approach used in interoperability module to be a very elegant solution:

Make a generic schema, e.g. `heightSchema` and then make 2 endpoint-specific aliases:

```
export const getDefaultRewardAtHeightRequestSchema = heightSchema;
export const whateverWeWillCallThisNewEndpointRequestSchema = heightSchema;
```

> The input height is needed to determine the current reward bracket.

If only the current reward bracket is needed [not a future one], then why is it needed to provide a param at all? Can't we somewhere down the line default height to current height? 🤔

Reply:
Updated to use alias per your suggestion.
For reward bracket maybe it's good to have the option to specify height to get future inflation rate as @shuse2 suggests.

Reply:
Updated the endpoint name to `getAnnualInflationRate` as discussed